### PR TITLE
bpo-31681: Make sure pkgutil.get_data() closes files properly

### PR DESCRIPTION
**Pull request by elprans**
https://bugs.python.org/issue31681

### CODE REVIEW

File: Lib/pkgutil.py [hunk 1]
```diff
@@ -252,7 +252,8 @@ def load_module(self, fullname):
         return mod
 
     def get_data(self, pathname):
-        return open(pathname, "rb").read()
+        with open(pathname, "rb") as file:
+            return file.read()
```
Comment (merwok) [inline Lib/pkgutil.py:253]: Top of the module says:

```
# NOTE: This module must remain compatible with Python 2.3, as it is shared
# by setuptools for distribution with Python 2.3 and up.
```

Would you mind investigating if that’s still true?

File: Misc/NEWS.d/next/Library/2017-10-03-15-41-08.bpo-31681.sOJMKV.rst [hunk 2]
```diff
@@ -0,0 +1 @@
+pkgutil.get_data() does not leak open files
```
Comment (merwok) [inline Misc/NEWS.d/next/Library/2017-10-03-15-41-08.bpo-31681.sOJMKV.rst]: Not sure if a news entry is useful.

**Review by merwok — CHANGES_REQUESTED** [review]

File: Lib/pkgutil.py (hunk 1 above)
Comment (merwok) [reply Lib/pkgutil.py:253]: (I don’t expect anyone to use versions older than 2.5, but who knows.  Just being extra cautious.)

Comment (1st1) [reply Lib/pkgutil.py:253]: I care way more about leaking resources than about compatibility with Python < 2.5. I believe that this isn't something we should be cautious about.

File: Misc/NEWS.d/next/Library/2017-10-03-15-41-08.bpo-31681.sOJMKV.rst (hunk 2 above)
Comment (1st1) [reply Misc/NEWS.d/next/Library/2017-10-03-15-41-08.bpo-31681.sOJMKV.rst]: It is, we want to have NEWS entries for virtually all commits we make.  Whether those entries will be highlighted in the 'What's new' document is a separate question.

**Comment by 1st1** [comment]
@merwok I'm going to merge this PR unless you have other arguments against it.

**Review by 1st1 — APPROVED** [review]

**Review by merwok — APPROVED** [review]

File: Lib/pkgutil.py (hunk 1 above)
Comment (merwok) [reply Lib/pkgutil.py:253]: I don’t disagree!  Maybe remove the comment in that case, so that future reviews don’t ask the same questions.

Comment (merwok) [reply Lib/pkgutil.py:253]: I checked setuptools and it doesn’t include pkgutil in the current version.  I removed the comment and would be happy to merge.

**Comment by merwok** [comment]
Thanks again for the fix!

**Pull request merged**